### PR TITLE
Upgrade to Tekton v0.14.3

### PR DESCRIPTION
**Pull request by rinckm**
Changes:
- TaskRun version goes from v1alpha1 to v1beta1
- Inputs.Params -> Params
- PodTemplate: tekton.PodTemplate -> *tekton.PodTemplate
- ReasonTimedOut (string) -> TaskRunReasonTimedOut (TaskRunReason)
- ReasonFailed (string) -> TaskRunReasonFailed (TaskRunReason)

Adjust termination message handling

### Dependency release notes

**SAP/stewardci-jenkinsfilerunner-image**
[Diff 200903_b053f5f...200921_6cc247f](https://github.com/SAP/stewardci-jenkinsfilerunner-image/compare/200903_b053f5f...200921_6cc247f)
- [200921_6cc247f](https://github.com/SAP/stewardci-jenkinsfilerunner-image/releases/tag/200921_6cc247f)

**tektoncd/pipeline**
[Diff v0.8.0...v0.14.3](https://github.com/tektoncd/pipeline/compare/v0.8.0...v0.14.3)

- [v0.14.3](https://github.com/tektoncd/pipeline/releases/tag/v0.14.3), [v0.14.2](https://github.com/tektoncd/pipeline/releases/tag/v0.14.2), [v0.14.1](https://github.com/tektoncd/pipeline/releases/tag/v0.14.1), [v0.14.0](https://github.com/tektoncd/pipeline/releases/tag/v0.14.0)
- [v0.13.2](https://github.com/tektoncd/pipeline/releases/tag/v0.13.2), [v0.13.1](https://github.com/tektoncd/pipeline/releases/tag/v0.13.1), [v0.13.0](https://github.com/tektoncd/pipeline/releases/tag/v0.13.0)
- [v0.12.1](https://github.com/tektoncd/pipeline/releases/tag/v0.12.1), [v0.12.0](https://github.com/tektoncd/pipeline/releases/tag/v0.12.0)
- [v0.11.3](https://github.com/tektoncd/pipeline/releases/tag/v0.11.3), [v0.11.2](https://github.com/tektoncd/pipeline/releases/tag/v0.11.2), [v0.11.1](https://github.com/tektoncd/pipeline/releases/tag/v0.11.1), [v0.11.0](https://github.com/tektoncd/pipeline/releases/tag/v0.11.0)
- [v0.10.1](https://github.com/tektoncd/pipeline/releases/tag/v0.10.1), [v0.10.0](https://github.com/tektoncd/pipeline/releases/tag/v0.10.0)
- [v0.9.2](https://github.com/tektoncd/pipeline/releases/tag/v0.9.2), [v0.9.1](https://github.com/tektoncd/pipeline/releases/tag/v0.9.1), [v0.9.0](https://github.com/tektoncd/pipeline/releases/tag/v0.9.0)

**knative.dev/pkg**
[Diff 1c6815d7e4c9...ea4d6e985ba0](https://github.com/knative/pkg/compare/1c6815d7e4c9...ea4d6e985ba0)

### Submitter checklist

- [x] Change has been tested (on a back-end cluster)
  System configuration:
  - controller images: TEST_200925_e2b1f7c
  - jfr image:  stewardci/stewardci-jenkinsfile-runner:200921_6cc247f
  - helm chart: 
    - steward-system: steward-0.5.0-dev (e2b1f7c)
    - tekton: tekton-0.14.3

  Test result: ok      github.com/SAP/stewardci-core/test/integrationtest
- [x] [changelog.yaml] with upgrade notes are prepared and appropriate for the audience affected by the change (users or developer, depending on the change).
- [x] Semantic version diffed against [last release][releases] and updated accordingly. In this project the version has to be maintained here:
  - [/charts/steward/Chart.yaml](https://github.com/SAP/stewardci-core/blob/master/charts/steward/Chart.yaml)

In case dependencies have been updated:
- [x] Links to external changelogs added since the last release of our component
- [x] Changelogs read thoroughly, potential impact described, upgrade notes prepared (if necessary)
- [x] Check if dependency update affects our semantic version increment.

### Reviewer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 1 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] The Pull Request title is understandable and reflects the changes well
- [x] The Pull Request description is understandable and well documented
- [x] 'Upgrade notes' are documented in changelog.yaml (if required)
- [x] [changelog.yaml] entry for this Pull Request has been added
  - [x] Changelog entry contains all required information

[changelog.yaml]: https://github.com/SAP/stewardci-core/changelog.yaml
[releases]: https://github.com/SAP/stewardci-core/releases

### CODE REVIEW

**Review by alxsap — COMMENTED** [review]
Looks good, but I did not read Tekton release notes and I did not test it yet.

File: docs/install/README.md [hunk 1]
```diff
@@ -3,14 +3,14 @@
 To run you own project "Steward" you need a Kubernetes cluster.
 Currently **Kubernetes 1.14** is recommended.
 
-## Install Tekton v0.8.0
+## Install Tekton v0.15.2
```
Comment (romanisb) [inline docs/install/README.md:6]: 0.15.2 or 0.14.3?

Comment (rinckm) [reply docs/install/README.md]: I tried first with 0.15.2 and then switched to 0.14.3. 
This is fixed in the README now.

File: changelog.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# THE CHANGELOG FORMAT IS BASED ON https://github.wdf.sap.corp/ConCISe/Concepts/blob/master/docs/Implementation/cloudci_backend_release_process/changelog.yaml
```
Comment (anbrsap) [inline changelog.yaml:1]: Reference to sap-internal resource -> remove!!!

**Review by anbrsap — COMMENTED** [review]
Names should be written in upper case.

File: go.mod [hunk 3]
```diff
@@ -34,11 +34,10 @@ require (
 	k8s.io/klog/v2 v2.1.0
 	k8s.io/legacy-cloud-providers v0.18.3 // indirect
 	k8s.io/utils v0.0.0-20200603063816-c1c6865ac451 // indirect
-	knative.dev/pkg v0.0.0-20200528142800-1c6815d7e4c9
+	knative.dev/pkg v0.0.0-20200702222342-ea4d6e985ba0
```
Comment (anbrsap) [inline go.mod:37]: Shouldn't we also upgrade to a newer `knative.dev/pkg` (see replace section)?

File: changelog.yaml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    description: >-
+      Build and run against Tekton v0.14.3.
+    warning: Requires Tekton v0.14.3.
+
+    upgradeNotes: >-
+      The target Kubernetes system must have Tekton v0.14.3 installed. Higher versions may also work.
+      In addition, Steward Jenkinsfile Runner Image version <a href="https://github.com/SAP/stewardci-jenkinsfilerunner-image/releases/tag/200921_6cc247f">200921_6cc247f</a> or later is required.
+      or higher.
+    pullRequest: 151
+    jiraIssue: 1330
```
Comment (alxsap) [inline changelog.yaml:32]: Internal. Do we want to keep it anyway? would not harm (if only the number is mentioned)...
But if we keep it as number we should rename the key to `jiraIssueNumber` (and `pullRequestNumber`.

File: go.mod (hunk 3 above)
Comment (rinckm) [reply go.mod:37]: Done

File: changelog.yaml (hunk 4 above)
Comment (rinckm) [reply changelog.yaml:32]: I would vote for keeping it as jiraIssueNumber.

File: changelog.yaml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+#      impact: [incompatible, minor, patch]
+#      title: <title>
+#      description: >-
+#        <description>
+#      warning: >-
+#        <optional warning message>
+#      upgradeNotes: >-
+#        <optional upgrade guidelines>
+#      pullRequest: <pull request number>
+#      jiraIssue: <Jira issue number in CLOUDCISERVICE>
```
Comment (alxsap) [inline changelog.yaml:14]:
```suggestion
#      jiraIssue: <Jira issue number>

```

File: changelog.yaml [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+#    - type: [bug, enhancement, internal]
+#      impact: [incompatible, minor, patch]
+#      title: <title>
+#      description: >-
+#        <description>
+#      warning: >-
+#        <optional warning message>
+#      upgradeNotes: >-
+#        <optional upgrade guidelines>
+#      pullRequest: <pull request number>
```
Comment (alxsap) [inline changelog.yaml:13]:
```suggestion
#      pullRequestNumber: <pull request number>

```

File: changelog.yaml [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+    description: >-
+      Build and run against Tekton v0.14.3.
+    warning: Requires Tekton v0.14.3.
+
+    upgradeNotes: >-
+      The target Kubernetes system must have Tekton v0.14.3 installed. Higher versions may also work.
+      In addition, Steward Jenkinsfile Runner Image version <a href="https://github.com/SAP/stewardci-jenkinsfilerunner-image/releases/tag/200921_6cc247f">200921_6cc247f</a> or later is required.
+      or higher.
+    pullRequest: 151
```
Comment (alxsap) [inline changelog.yaml:31]:
```suggestion
    pullRequestNumber: 151

```

File: changelog.yaml (hunk 4 above)
Comment (alxsap) [inline changelog.yaml:32]:
```suggestion
    jiraIssueNumber: 1330

```

File: changelog.yaml (hunk 2 above)
Comment (rinckm) [reply changelog.yaml]: Done

File: changelog.yaml [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+#      jiraIssue: <Jira issue number in CLOUDCISERVICE>
+
+- version: "NEXT"
+  date: TBD
+  changes:
+  - type: enhancement
+    impact: incompatible
+    title: Upgrade tekton dependency to version v0.14.3
+    description: >-
+      Upgrade tekton dependency to version v0.14.3 and adjust controller accordingly.
```
[the remark this answers is not in the document]
Comment (rinckm) [reply changelog.yaml]: Done

File: changelog.yaml [hunk 9]
```diff
@@ -0,0 +1,35 @@
+## ----------------------------------
```
Comment (alxsap) [inline changelog.yaml:1]: See also https://github.com/SAP/stewardci-core/pull/159

**Review by alxsap — COMMENTED** [review]
Looks good, but I have no time to dig deeper and test it.

File: changelog.yaml [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+    title: Upgrade dependency to Tekton v0.14.3
+
+    description: >-
+      Build and run against Tekton v0.14.3.
+    warning: Requires Tekton v0.14.3.
+
+    upgradeNotes: >-
+      The target Kubernetes system must have Tekton v0.14.3 installed. Higher versions may also work.
+      In addition, Steward Jenkinsfile Runner Image version <a href="https://github.com/SAP/stewardci-jenkinsfilerunner-image/releases/tag/200921_6cc247f">200921_6cc247f</a> or later is required.
+      or higher.
```
Comment (anbrsap) [inline changelog.yaml:32]:
```suggestion

```

File: pkg/runctl/run.go [hunk 11]
```diff
@@ -62,6 +62,34 @@ func (r *tektonRun) IsFinished() (bool, steward.Result) {
...
+		allMessages, err := termination.ParseMessage(msg)
+		if err == nil {
+			for _, singleMessage := range allMessages {
+				if singleMessage.Key == jfrResultKey {
+					return singleMessage.Value
+				}
+			}
+		} else {
+			return msg
+		}
```
Comment (anbrsap) [inline pkg/runctl/run.go:88]: Put error handling code first and save the additional indentation for the 'normal' flow:
```suggestion
		if err != nil {
			return msg
		}
		for _, singleMessage := range allMessages {
			if singleMessage.Key == jfrResultKey {
				return singleMessage.Value
			}
		}
```

**Review by anbrsap — APPROVED** [review]

**Pull request merged**